### ISSUE TITLE
Getting the saltNonce

### ISSUE THREAD

**Issue by ghost** [nlordell/deadbeef#10]
Hi - trying to use this generator, but I don't understand where the saltNonce is coming from?

Shouldn't there be one more output after callData?

**Comment by ghost** [comment]
I added a println to line 125:

`println!("salt_nonce:  0x{}", hex::encode(&safe.salt_nonce()));`

Which gave me the following:

```
deadbeef --owner 0x1fE926205440d6A61119d231FA28e1519514E2E5 --prefix 0x11
address:   0x11B3Ff10303D91Ba30Cda49dFf95FbE02c4cC257
factory:   0x4e1DCf7AD4e460CfD30791CCC4F9c8a4f820ec67
singleton: 0x41675C099F32341bf84BFc5382aF534df5C7461a
fallback:  0xfd0732Dc9E303f09fCEf3a7388Ad10A83459Ec99
owners:    0x1fE926205440d6A61119d231FA28e1519514E2E5
threshold: 1
calldata:  0x1688f0b900000000000000000000000041675c099f32341bf84bfc5382af534df5c7461a0000000000000000000000000000000000000000000000000000000000000060ad1ae6aa9949e7cf639d106cd1663fb2827b7825f1f075f90cce816fae60df370000000000000000000000000000000000000000000000000000000000000164b63e800d0000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000140000000000000000000000000fd0732dc9e303f09fcef3a7388ad10a83459ec9900000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000001fe926205440d6a61119d231fa28e1519514e2e5000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
salt_nonce:  0xad1ae6aa9949e7cf639d106cd1663fb2827b7825f1f075f90cce816fae60df37
```

However, when I put those into the `createProxyWithNonce` on the factory contract, the contract address does not match the address output:

https://goerli.etherscan.io/tx/0x5d561fc0cb51265a29af4281a057ef986de44cf73f4e1eadea1ae44541cb03e4

**Comment by nlordell** [comment]
Hey, thanks for posting this - I think it is a documentation issue that can be improved in the repository:

>  but I don't understand where the saltNonce is coming from?

The `saltNonce` is already included in the `calldata`. In fact, the `calldata` in the output is an encoded `createProxyWithNonce` call and what should be used with a transaction to the proxy factory directly (and not the `initializer` bytes).

An easy way to extract the parameters to the `createProxyWithNonce` call is to use something like this [ETH Calldata Decoder](https://calldata-decoder.apoorv.xyz) and paste in the `calldata` there. You would get:

```json
{
  "allPossibilities": [
    {
      "function": "createProxyWithNonce(address,bytes,uint256)",
      "params": [
        "0x41675C099F32341bf84BFc5382aF534df5C7461a",
        "0xb63e800d0000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000140000000000000000000000000fd0732dc9e303f09fcef3a7388ad10a83459ec9900000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000001fe926205440d6a61119d231fa28e1519514e2e50000000000000000000000000000000000000000000000000000000000000000",
        "78297652829583202014385281856584591471113668308889242546575654979373178609463"
      ]
    }
  ]
}
```

Where the parameters are:
- `singleton`: `0x41675C099F32341bf84BFc5382aF534df5C7461a`
- `initializer`: `0xb63e800d...`
- `saltNonce`: `78297652829583202014385281856584591471113668308889242546575654979373178609463` which is equivalent to what you found:

```bash
$ python3 -c 'print(0xad1ae6aa9949e7cf639d106cd1663fb2827b7825f1f075f90cce816fae60df37 == 78297652829583202014385281856584591471113668308889242546575654979373178609463)'
True
```

Now, if you execute with this in mind, you get the expected address:

https://goerli.etherscan.io/tx/0xa4ad99f229c6adcca34afeebcf3b8271ed854f7bbb18e3af673490a34719c64e

**Comment by nlordell** [comment]
In particular, this section from the README can be expanded: https://github.com/nlordell/deadbeef#creating-the-safe

I'm leaving this issue open until the documentation is improved. I'll also consider adding a `--verbose` flag to print the salt nonce and initializer bytes.

**Comment by ghost** [comment]
I am officially working too much it seems.

I've suggested a tweak to the documentation here #11

**Comment by nlordell** [comment]
Addressed in #11
